### PR TITLE
Create another offer if one gets taken

### PR DESCRIPTION
**Pull request by klochowicz**
See details in commit messages.

### CODE REVIEW

File: daemon-tests/tests/happy_path.rs
Comment (thomaseizinger) [inline daemon-tests/tests/happy_path.rs]:
```suggestion
async fn another_offer_is_automatically_created_after_taker_takes_order() {
```

File: daemon-tests/tests/happy_path.rs
Comment (thomaseizinger) [inline daemon-tests/tests/happy_path.rs]: We might want to rename this to something like `set_offer_parameters`?

**Review by thomaseizinger — APPROVED** [review]

File: daemon-tests/tests/happy_path.rs
Comment (klochowicz) [reply daemon-tests/tests/happy_path.rs]: sounds good, changed this and also changed `NewOrder` to `OfferParams`

**Review by thomaseizinger — APPROVED** [review]
I like this change but it might be controversial so please gather some more approvals before merging :)

**Comment by klochowicz** [comment]
> I like this change but it might be controversial so please gather some more approvals before merging :)

no worries! will wait till Monday, where I'll mention it on the daily.

For other reviewers, a short summary: 
 - this change adjusts the mechanism of creating the offers in the maker
 - instead of manually creating "each" offer, we essentially send a feed of latest parameters from which offers are constructed
 - this change is a preparation for the plan of sending `price_long` as one more parameter (a price for the offers where makers goes long), which will allow for simultaneous creation of offers "the other way". 
 - as a side effect, we'll be able to pump more offers with the params we want more quickly 😃 

a bit more on future direction:
- I'd like to explore allowing setting the `price_long` and `price_short` as `Option<>`, so that the maker could have a means to choose whether they want to go both long and short, just one way, or none (and just allow rolling over currently active CFDs).

**Comment by thomaseizinger** [comment]
I think this goes back on ADR003. Can you please draft a short ADR that explains your reasoning?

**Review by bonomat — APPROVED** [review]
Let's give this a try +1

**Comment by klochowicz** [comment]
> I think this goes back on ADR003. Can you please draft a short ADR that explains your reasoning?

I had a re-read of ADR003 and I don't think it goes against it. The same amount of control of the business logic will be available (including stopping new orders from happening, when I introduce `Option<Price>` shortly). is there a specific part that raises your concerns?

edit: perhaps more broadly, I understand this change like a difference in "ergonomics" of using the API, not as a change of the business logic. exactly the same strategies are available to whoever operates the maker.

**Comment by thomaseizinger** [comment]
> > I think this goes back on ADR003. Can you please draft a short ADR that explains your reasoning?
> 
> I had a re-read of ADR003 and I don't think it goes against it. The same amount of control of the business logic will be available (including stopping new orders from happening, when I introduce `Option<Price>` shortly). is there a specific part that raises your concerns?
> 
> edit: perhaps more broadly, I understand this change like a difference in "ergonomics" of using the API, not as a change of the business logic. exactly the same strategies are available to whoever operates the maker.

The decision logic for sending out an offer changes. Previously it was "every time a new offer gets created on the API" and now it is "every time an offer gets taken". It doesn't contract the original ADR but what we are doing now was not the original idea so I think it is worth clarifying that. It is not a big decision but it would be good to record it.

It is also a learning for the future in how such a system should/could be designed.

**Comment by klochowicz** [comment]
> > > I think this goes back on ADR003. Can you please draft a short ADR that explains your reasoning?
> > 
> > 
> > I had a re-read of ADR003 and I don't think it goes against it. The same amount of control of the business logic will be available (including stopping new orders from happening, when I introduce `Option<Price>` shortly). is there a specific part that raises your concerns?
> > edit: perhaps more broadly, I understand this change like a difference in "ergonomics" of using the API, not as a change of the business logic. exactly the same strategies are available to whoever operates the maker.
> 
> The decision logic for sending out an offer changes. Previously it was "every time a new offer gets created on the API" and now it is "every time an offer gets taken". It doesn't contract the original ADR but what we are doing now was not the original idea so I think it is worth clarifying that. It is not a big decision but it would be good to record it.
> 
> It is also a learning for the future in how such a system should/could be designed.

I take your point in a valuable learning opportunity. I've drafted a quick ADR here:
https://github.com/itchysats/architecture/pull/14

Will merge this in the meantime as I consider that I got enough buy-in to try this approach (that will unblock taker going short too).

**Pull request merged**
